packages/nameguard-sdk/src/index.ts

```diff
@@ -372,27 +376,50 @@ export interface ConsolidatedNameGuardReport extends ConsolidatedReport {
 /**
  * NameGuard report that contains the full results of all `checks` on all `labels` in a name.
  */
-export interface NameGuardReport extends ConsolidatedNameGuardReport {
+export interface AbstractNameGuardReport extends ConsolidatedNameGuardReport {
+  /* Whether or not the name was inspected. */
+  inspected: boolean;
+
   /* The results of all checks performed by NameGuard on `name`. */
   checks: CheckResult[];
 
-  /** Details of the inspection of all labels in `name`. */
-  labels: LabelGuardReport[];
+  /** Details of the inspection of all labels in `name`.
+    *
+    * `undefined` if `name` was uninspected and this is an `UninspectedNameGuardReport`.
+  */
+  labels?: LabelGuardReport[];
 
   /**
    * The name considered to be the canonical form of the analyzed `name`.
    *
-   * `null` if and only if the canonical form of `name` is considered to be undefined.
+   * `undefined` if and only if the canonical form of `name` is considered to be undefined or `name` was uninspected and this is an ``UninspectedNameGuardReport`.
    *
    * If a label is represented as `[labelhash]` in `name`,
    * the `canonical_name` will also contain the label represented as `[labelhash]`.
    *
    * `canonical_name` is guaranteed to be normalized with the exception of the case
    * where `normalization` is `unknown` and one or more labels are represented as `[labelhash]`.
    */
-  canonical_name: string | null;
+  canonical_name?: string;
+}
+
+export interface InspectedNameGuardReport extends AbstractNameGuardReport {
+    labels: LabelGuardReport[];
+    inspected: true;
 }
 
+/**
+ * NameGuard report for a name that was exceptionally long and was not inspected for performance reasons.
+ * Fields `label` and canonical_name` are undefined, `rating` is `alert`, `highest_risk` is `uninspected`, `risk_count` is `1`.
+ */
+export interface UninspectedNameGuardReport extends AbstractNameGuardReport {
+    labels: undefined;
+    canonical_name: undefined;
+    inspected: false;
+}
+
+export type NameGuardReport = InspectedNameGuardReport | UninspectedNameGuardReport;
```

Comment:
@notrab Hey, please check how we've defined the types here for `NameGuardReport`, `InspectedNameGuardReport`, and `UninspectedNameGuardReport`.

The goal for doing this was to make it so that in the case you need to support both the inspected and uninspected cases you can define a field as type `NameGuardReport`. Then, if you write some code where you do something like:

```
const myFunction = (report: NameGuardReport) => void {
   if (report.inspected) {
      // Here TypeScript will know that `report` is of type `InspectedNameGuardReport`.
   } else {
      // Here TypeScript will know that `report` is of type `UninspectedNameGuardReport`.
   }
}
```

This means we might have cases in our code where we should not define a variable as `NameGuardReport` but instead might want to define it as type `InspectedNameGuardReport`, such that the type system enforces us to first verify that the report we're passing in was indeed inspected.

Can you please investigate and confirm that all code we have in `nameguard-react`, `examples.nameguard.io` or `nameguard.io` is updated based on this concept?